### ISSUE TITLE
Clarify the difference between `pipe` and `pipe/async`

### ISSUE THREAD

**Issue by lambdalisue** [jsr-core/pipe#6]
Explain the difference in English with maybe some code like

```ts
import { pipe } from "jsr:@core/pipe";
import { pipe as pipeAsync } from "jsr:@core/pipe/async";

const a = (v: unknown) => v;
const b = (v: unknown) => v;
const c = (v: unknown) => v;

// Equivalent
console.log(pipe(1, a, b, c)); // 1
console.log(c(b(a(1)))); // 1

// Equivalent
console.log(await pipeAsync(1, a, b, c)); // 1
console.log(await c(await b(await a(1)))); // 1
```

**Comment by lambdalisue** [comment]
The original issue was #3

**Comment by lambdalisue** [comment]
Closed by #7
